examples/networkgraph.cpp

```diff
@@ -0,0 +1,909 @@
+
+    void set_fixed_pres(double pres){m_fixed_pres = pres; };
+    double get_fixed_pres(){return m_fixed_pres; };
+
+    int get_xpos(){return m_xpos;};
+
+    void print(std::stringstream& netw_str);
+
+    bool detete_from_inlet_list(const std::string& name);
+
+    bool add_well(const std::string& name);
+
+    void reset_outlet(){m_outlet = nullptr;};
+
+private:
+    int m_vfp;
+    int m_xpos = -1;
+    double m_fixed_pres = -1.0;
+    int m_ind = -1;
+
+    std::string m_name;
+    std::shared_ptr<Node> m_outlet;
+    std::vector<std::shared_ptr<Node>> m_inlet_list;
+    std::vector<std::string> m_well_list;
+
+    std::shared_ptr<Node> next_branch();
+};
+
+
+Node::Node(const std::string& name)
+   : m_name(name)
+{
+    m_vfp = -1;
+}
+
+std::shared_ptr<Node> Node::next_branch(){
+
+    std::shared_ptr<Node> p1 = m_outlet;
+
+    while (p1 != nullptr){
+        for (size_t n = 0; n < p1->m_inlet_list.size(); n ++){
+            if (p1->m_inlet_list[n]->m_xpos == -1){
+                return p1;
+            }
+        }
+        p1 = p1->m_outlet;
+    }
+
+    return p1;
+}
+
+bool Node::detete_from_inlet_list(const std::string& name){
+
+    int ind = -1;
+
+    for (size_t n = 0; n < m_inlet_list.size(); n ++)
+        if (m_inlet_list[n]->name() == name)
+            ind = n;
+
+    if (ind < 0){
+        return false;
+    } else {
+        m_inlet_list.erase (m_inlet_list.begin() + ind);
+        return true;
+    }
+}
+
+bool Node::add_well(const std::string& name){
+   if (m_inlet_list.size() > 0)
+       return false;
+
+   m_well_list.push_back(name);
+
+   return true;
+}
+
+
+void Node::print(std::stringstream& netw_str){
+
+    netw_str.seekg(0, std::ios::end);
+    int init_length = netw_str.tellg();
+
+    int pos_lineshift = 0;
+
+    auto p = netw_str.str().find_last_of("\n");
+
+    if (p != std::string::npos)
+        pos_lineshift = p + 1;
+
+    if (m_outlet == nullptr){
+        netw_str << "  o (" << m_name << ")";
+        m_xpos = netw_str.str().find_first_of("o", init_length) - pos_lineshift;
+    } else if (m_vfp == 9999){
+        netw_str << " --- +(" << m_name << ")";
+        m_xpos = netw_str.str().find_first_of("+", init_length) - pos_lineshift;
+    } else {
+        netw_str << " --[" << m_vfp << "]-- +(" << m_name << ")";
+        m_xpos = netw_str.str().find_first_of("+", init_length) - pos_lineshift;
+    }
+
+
+    for (size_t m = 0; m < m_inlet_list.size(); m ++)
+        m_inlet_list[m]->print(netw_str);
+
+    if (m_inlet_list.size() == 0){
+        auto next_br = next_branch();
+
+        netw_str << " : ";
+
+        for (auto& well: m_well_list)
+            netw_str << " " << well;
+
+        if (next_br != nullptr){
+            std::string xpos_str(next_br->m_xpos, ' ');
+            netw_str << "\n" << xpos_str << "\\\n" << xpos_str;
+        }
+    }
+}
+
+void Node::add_inlet_node(std::shared_ptr<Node> node)
+{
+    for (size_t n = 0; n < m_inlet_list.size(); n ++ ){
```

Comment:
I do not understand this code. Why do you erase and then push?

If that is indeed what you need to do, the code is also a bit hard to maintain, for example continuing to loop and not doing the break would skip an element. It is better to use std::find() to find the matching node and erase that (if not at the end).

Reply:
updated with std::find_if. Yes this is needed. Consider following example 

```
BRANPROP
-- downtree   uptree
-- node       node
  'PD1'      'PROD1'  8   /
  'PD1'      'PROD1'  9   /
  'TEST1'    'PD1'  9999 /
/
```

The second record is replacing the the first and changing the lift curve.